hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/runtime/DockerLinuxContainerRuntime.java

```diff
@@ -589,13 +601,17 @@ public void launchContainer(ContainerRuntimeContext ctx)
     Map<String, String> environment = container.getLaunchContext()
         .getEnvironment();
     String imageName = environment.get(ENV_DOCKER_CONTAINER_IMAGE);
```

Comment:
does it mean that we still allow the user to select its desired image ?

Reply:
As we won't allow this env variable to be overridden, users won't be able to choose.
This env variable will directly come from admin-env.ENV_DOCKER_CONTAINER_IMAGE which is our way to define the default image